src/main/java/xyz/niflheim/stockfish/engine/enums/Query.java

```diff
@@ -93,6 +99,34 @@ public long getMovetime() {
         return movetime;
     }
 
+    public void setDepth(int depth) {
+        this.depth = depth;
+    }
+
+    public void setMovetime(int movetime) {
+        this.movetime = movetime;
+    }
+
+
+    private boolean isWithinLimits() {
+        if (getType().equals(Best_Move) && ((getDepth() < 0 || getMovetime() < 0)
```

Comment:
Here it is advisable to do Best_Move.equals(getType()) to ensure null safety

Reply:
Implemented as you requested